site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/known customer credential/knownCustomerCredentialIssuer.test.js

```diff
@@ -0,0 +1,499 @@
+    credential_configurations_supported: {
+      // type of credentials Issuer supports and what credential will look like
+      KnownCustomerCredential: {
+        format: 'jwt_vc_json',
+        scope: 'CustomerIdentity',
+        cryptographic_binding_methods_supported: ['did:example'],
+        credential_signing_alg_values_supported: ['ES256'],
+        credential_definition: {
+          type: ['VerifiableCredential', 'KnownCustomerCredential'],
+          credentialSubject: {
+            country: {
+              display: [{ name: 'Country', locale: 'en-US' }],
+            },
+          },
+        },
+        proof_types_supported: {
+          jwt: {
+            proof_signing_alg_values_supported: ['ES256'],
+          },
+        },
+        display: [
+          {
+            name: 'Known Customer Credential',
+            locale: 'en-US',
+            logo: {
+              url: 'https://issuer.example.com/public/logo.png',
+              alt_text: 'Issuer Logo',
+            },
+            background_color: '#FFFFFF',
+            text_color: '#000000',
+          },
+        ],
+      },
+    },
+  };
+
+  res.json(issuerMetadata);
+});
+// :snippet-end:
+
+// :snippet-start: wellKnownAuthorizationServerEndpointIssuerJs
+app.get('/.well-known/oauth-authorization-server', (req, res) => {
+  const oauthAuthorizationServerMetadata = {
+    issuer: 'https://issuer.example.com', // URL of the Credential Issuer
+    token_endpoint: 'https://issuer.example.com/token', // URL for the Access Token Request
+  };
+
+  res.json(oauthAuthorizationServerMetadata);
+});
+// :snippet-end:
+
+// :snippet-start: accessTokenEndpointIssuerJs
+import { VerifiableCredential, Jwt } from '@web5/credentials';
+
+const accessTokenToCNonceMap = new Map();
+
+app.post('/token', async (req, res) => {
+  const { grant_type, code } = req.body;
+
+  if (grant_type !== 'urn:ietf:params:oauth:grant-type:pre-authorized_code') {
+    return res.status(400).json({ error: 'unsupported_grant_type' });
+  }
+
+  const customersDidUri = preAuthCodeToDidMap.get(code);
+  if (!customersDidUri) {
+    return res.status(400).json({ error: 'invalid_grant' });
+  }
+
+  /*******************************************
+     Create the payload for the access token
+    ********************************************/
+  const accessTokenPayload = {
+    sub: customersDidUri, // Customer's DID string
+    iss: issuersBearerDid.uri, // Issuer's DID string
+    iat: Math.floor(Date.now() / 1000), // Issued at
+    exp: Math.floor(Date.now() / 1000) + 86400, // Expiration time
+  };
+  /*******************************************
+     sign accessToken and generate a c_nonce
+    ********************************************/
+  try {
+    const accessToken = await Jwt.sign({
+      signerDid: issuerBearerDid,
+      payload: accessTokenPayload,
+    });
+
+    const cNonce = generateCNonce();
+    accessTokenToCNonceMap.set(accessToken, cNonce);
+
+    res.json({
```

Comment:
We also need the issuer to invalidate the pre auth code upon successful issuance of the access token

```suggestion
    accessTokenToCNonceMap.set(accessToken, cNonce);

    preAuthCodeToDidMap.delete(code)

    res.json({
```